pkg/network/protocols/postgres/telemetry.go

```diff
@@ -97,3 +98,131 @@ func (t *Telemetry) Count(tx *ebpf.EbpfEvent, eventWrapper *EventWrapper) {
 func (t *Telemetry) Log() {
 	log.Debugf("postgres stats summary: %s", t.metricGroup.Summary())
 }
+
+// PgKernelTelemetry	provides empirical kernel statistics about the number of messages in each TCP packet
+type PgKernelTelemetry struct {
+	// metricGroup is used here mostly for building the log message below
+	metricGroup *libtelemetry.MetricGroup
+	// pgMessagesCountBuckets		count Postgres messages and divide the count into buckets
+	pgMessagesCountBuckets [pgebpf.KerMsgCountNumBuckets]*libtelemetry.TLSAwareCounter
+	// pgMessagesCountNotFound		count cases when message was not detected in packet
+	pgMessagesCountNotFound *libtelemetry.TLSAwareCounter
+	// pgMessagesCountAboveMax		count cases when maximum number of messages was exceeded
+	pgMessagesCountAboveMax *libtelemetry.TLSAwareCounter
+	// pgMessagesCountErrors		count cases where an error occurred while counting messages
+	pgMessagesCountErrors *libtelemetry.TLSAwareCounter
+	// pgKernelMsgCountPlainPrev	save the last counters observed from the kernel, plain messages
+	pgKernelMsgCountPlainPrev pgebpf.PostgresKernelMsgCount
+	// pgKernelMsgCountPrev			save the last counters observed from the kernel, TLS messages
+	pgKernelMsgCountTlsPrev pgebpf.PostgresKernelMsgCount
+}
+
+// newPgKernelTelemetry this is the Postgres message counter store.
+func newPgKernelTelemetry() *PgKernelTelemetry {
+	metricGroup := libtelemetry.NewMetricGroup("usm.postgres_kernel", libtelemetry.OptStatsd)
```

Comment:
I think all metrics (ebpf or user mode) should be under `usm.postgres`
the metric name can specify more information

Reply:
changed